internal/core/src/indexbuilder/index_c.cpp

```diff
@@ -111,76 +96,58 @@ BuildBinaryVecIndex(CIndex index, int64_t data_size, const uint8_t* vectors) {
 // TODO: optimize here if necessary.
 CStatus
 BuildScalarIndex(CIndex c_index, int64_t size, const void* field_data) {
-    auto status = CStatus();
     try {
         AssertInfo(c_index, "failed to build scalar index, passed index was null");
 
         auto real_index = reinterpret_cast<milvus::indexbuilder::IndexCreatorBase*>(c_index);
         const int64_t dim = 8;  // not important here
         auto dataset = knowhere::GenDataset(size, dim, field_data);
         real_index->Build(dataset);
-
-        status.error_code = Success;
-        status.error_msg = "";
+        return milvus::SuccessCStatus();
     } catch (std::exception& e) {
-        status.error_code = UnexpectedError;
-        status.error_msg = strdup(e.what());
+        return milvus::FailureCStatus(UnexpectedError, e.what());
     }
-    return status;
 }
 
 CStatus
 SerializeIndexToBinarySet(CIndex index, CBinarySet* c_binary_set) {
-    auto status = CStatus();
     try {
         AssertInfo(index, "failed to serialize index to binary set, passed index was null");
         auto real_index = reinterpret_cast<milvus::indexbuilder::IndexCreatorBase*>(index);
         auto binary = std::make_unique<knowhere::BinarySet>(real_index->Serialize());
         *c_binary_set = binary.release();
-        status.error_code = Success;
-        status.error_msg = "";
+        return milvus::SuccessCStatus();
     } catch (std::exception& e) {
-        status.error_code = UnexpectedError;
-        status.error_msg = strdup(e.what());
+        return milvus::FailureCStatus(UnexpectedError, e.what());
     }
-    return status;
 }
 
 CStatus
 LoadIndexFromBinarySet(CIndex index, CBinarySet c_binary_set) {
-    auto status = CStatus();
     try {
         AssertInfo(index, "failed to load index from binary set, passed index was null");
         auto real_index = reinterpret_cast<milvus::indexbuilder::IndexCreatorBase*>(index);
         auto binary_set = reinterpret_cast<knowhere::BinarySet*>(c_binary_set);
         real_index->Load(*binary_set);
-        status.error_code = Success;
-        status.error_msg = "";
+        return milvus::SuccessCStatus();
     } catch (std::exception& e) {
-        status.error_code = UnexpectedError;
-        status.error_msg = strdup(e.what());
+        return milvus::FailureCStatus(UnexpectedError, e.what());
     }
-    return status;
 }
 
 CStatus
 QueryOnFloatVecIndex(CIndex index, int64_t float_value_num, const float* vectors, CIndexQueryResult* res) {
-    auto status = CStatus();
     try {
-        auto cIndex = (milvus::indexbuilder::VecIndexCreator*)index;
+        auto cIndex = reinterpret_cast<milvus::indexbuilder::VecIndexCreator*>(index);
```

Comment:
should be discuess: use reinterpret_cast or static_cast

Reply:
`reinterpret_cast` here is better, since you have already known this pointer point to `milvus::indexbuilder::VecIndexCreator`.